sycl/test/aie/ranged_buffer2.cpp

```diff
@@ -0,0 +1,34 @@
+// REQUIRES: aie
+
+// RUN: %aie_clang %s -o %t.bin
+// RUN: %if_run_on_device %run_on_device %t.bin > %t.check 2>&1
+// RUN: %if_run_on_device FileCheck %s --input-file=%t.check
+
+#include "aie.hpp"
+#include <numeric>
+
+int main() {
+  constexpr std::size_t size = 10;
+  aie::device<1, 1> dev;
+  aie::buffer<int> buff(10, 0);
+  std::iota(buff.begin(), buff.begin() + 5, 0);
+  for (int i = 0; i < size; i++) {
+    std::cout << "buff[" << i << "]=" << buff[i] << std::endl;
+  }
+  aie::queue q(dev);
+  q.submit_uniform([&](auto& ht) {
+    aie::accessor acc = aie::buffer_range(ht, buff)
+                            .read_range(5, buff.size())
+                            .write_range(5, buff.size());
+    ht.single_task([=](auto& dt) {
+      for (int i = 0; i < acc.size(); i++) {
+        acc[i] = i + 5;
```

Comment:
I am not sure to understand the code.
I was thinking to something more along
```suggestion
                            .read_range(0, buff.size()/2)
                            .write_range(5, buff.size()/2);
    ht.single_task([=](auto& dt) {
      for (int i = 0; i < acc.size()/2; i++) {
        acc[i + 5] = acc[i] + 5;
```
Or perhaps I do not understand your API?

Reply:
I added support to the aie++ lib from write ranges that are not a part of the read range. then to added a new test that is very close to the edit you requested.

important points:
 - ranges are `(start, end)` not `(start, size)`
 - ranges are absolute.
 - indexes on the device are relative to range of data on the device. the range on the device is `(min(read_start, write_start), max(read_end, write_end))`

Reply:
Move this information in the header of the implementation too.